lib/server.js

```diff
@@ -179,34 +179,30 @@ function handleResponse(opts, req, resp, response) {
         var body;
         // Prepare error responses for the client
         if (response.status >= 400) {
-            if (!response.body) {
-                response.body = {};
-            }
+            rh['content-type'] = rh['content-type']
+                || 'application/problem+json';
+
             // Whitelist fields to avoid leaking sensitive info
-            var rBody = response.body;
+            var rBody = response.body || {};
             body = {
                 type: rBody.type,
                 title: rBody.title,
-                method: rBody.method,
+                method: rBody.method || req.method,
                 detail: rBody.detail || rBody.description,
-                uri: rBody.uri
+                instance: rBody.instance || rBody.uri || req.uri
```

Comment:
The intention is not to link to the resource that produced the issue, but to a resource that "identifies the specific occurrence of the problem. It may or may not yield further information if dereferenced."